src/components/ContactDetails/index.tsx

```diff
@@ -3,13 +3,26 @@ import {Component} from 'react';
 import {observer} from 'mobx-react';
 
 import Contact from '../../interfaces/Contact';
-
 import {ProfilePicture} from './ProfilePicture';
+import {appState} from '../..';
 
 @observer
-export class ContactDetails extends Component<{selectedContact: Contact}, {}> {
+export class ContactDetails extends Component<{params: {contactId: string}}, {}> {
+
+  contact: Contact;
+
+  constructor(props){
+    super(props);
+
+    if (this.props.params.contactId) {
```

Comment:
Should be like:

```javascript
  componentWillMount() {
    if (this.props.params.contactId) {
      appState.setSelectedContactId(this.props.params.contactId);
    }

    this.contact = appState.selectedContact;
  }
```

See also: https://github.com/acdlite/redux-router/issues/171#issuecomment-155570297